File: 07 hashtables/exercise - firstRecurring.js
const firstRecurring = function (nums) {
  let set = new Set();
  for (let i = 0; i < nums.length; i++) {
    let itm = nums[i];

    if (set.has(itm))
      return itm;

    set.add(itm);
  }

  return undefined;
}

File: 07 hashtables/exercise - firstRecurring.cs
public int? firstRecurring(int[] nums)
{
  List<int> set = new List<int>(nums.Length);
  for (int i = 0; i < nums.Length; i++)
  {
    int itm = nums[i];

    if (set.Contains(itm))
      return itm;

    set.Add(itm);
  }

  return null;
}